### ISSUE TITLE
Gather a list of features and bugs we fixed in 2018.3

### ISSUE THREAD

**Issue by onox** [c172p-team/c172p#1108]
We should gather a list of all the new features, major changes, and bugs we fixed. Please create a list (use \* to create bullet points) here.

**Features**:

* Improved 3D models and textures of all levers, toggles, seats, magneto keys, EGT gauge and attitude indicator
* New recess casings to all panel instruments
* Improved labels of panel texture 
* Added 3D models to cockpit:
    - Alternate static source knob
    - Glove pocket to panel (holds GPS device when not in use)
    - Lighter hole
    - Overhead panel light switches for red flood, gauge post, and white dome/courtesy lights
    - Low Voltage LED
    - Sunvisors
    - PPT cables connected to the yokes
* New ammeter gauge matching the model used in the 172P
* Light maps for red flood, gauge post and white dome light
* ALS procedural lights (glare) for red flood, gauge post, white dome and wing courtesy light
* Light map illumination and procedural glare responds to available light and dims during daylight
* Re-positioned and improved 3D models and textures of all levers
* Added ambient occlusion map to all interior textures
* Added glass effect for gauges
* New sounds, including when clicking on the checklist, adding/removing control lock, mounting/dismounting GPS, opening/closing glove pocket, opening/closing the window latches, extending/retracting the water rudder cable
* Better sound for flaps lever and motor
* Interlocked master switch (BAT and ALT)
* GPS antenna is now up by default
* GPS night mode
* Amphibious gear advisor activated
* New amphibious gear lever
* Expanded help tool tips
* Add subtle glass reflection to panel instruments and radio stack
* Moved tutorials to Hilo Airport (PHTO)
* Improved tutorials and added two new ones: take off and landing for float variants
* Much improved propeller model
* Improved handle animation of doors and baggage door
* Doors should now be closed from the inside using the door strap
* Air resistance will force an open door or baggage door backwards
* Improved KMA20's marker sensitivity toggle animation
* Much improved vertical stabilizer model, including a retopologized beacon model
* Added external antennas for the ELT, transponder, ADF and marker beacons
* Removed wire connecting the wings to the tail as our ADF receiver uses a different type of antenna
* Animated water rudder cable
* New digital clock added
* New five slot Save and Resume feature
* QT launcher variant selection support
* QT launcher Location support including "On approach"

**Fixes**:

* Most of the cases of multiple objects mapped to a single texture have now been solved by assigning each their own texture
* 3D models of rotary knobs of VOR and NAV radios now turn 360 degrees
* Corrected the label of the OBS knob of the heading indicator
* Implemented a particle effect color and light manager for smoke and spray
* Adjusted P-factor effect
* Strobe lights are stronger now
* Fixed issue which caused the landing light to appear green under certain situations
* Removed shadow effect from radio stack readouts
* Autopilot readouts now match the rest of the radio stack in both colour and intensity
* Added a black panel behind the broken wing structure

People who have contributed to the 2018.3 release (`git shortlog version/2018.2..HEAD`):

* TODO

Do not forget to update `NEWS.md` before creating the release. See https://github.com/c172p-team/c172p-detailed/wiki/Release-process-and-versioning

**Comment by gilbertohasnofb** [comment]
@wlbragg please check if I did not miss anything or if you want to re-word, add or remove anything

**Comment by onox** [comment]
I've simplified the list a bit

**Comment by wlbragg** [comment]
I rearrange a couple things, like light maps and procedural glare out of 3d objects and into their own categories.

**Comment by gilbertohasnofb** [comment]
@wlbragg I just saw your message in the devel list, please do let us know here what 'deadline' you decide together with James so that we can finish all these new features by then. I would say 2-3 weeks should suffice and we will religiously stick to that. What you guys think?

**Comment by wlbragg** [comment]
That's fine with me, we already have enough for a nice, feature packed release. I guess we'll know more on timing when James gets back for vacation.

**Comment by wkitty42** [comment]
On 07/03/2018 01:35 PM, wlbragg wrote:
> That's fine with me, we already have enough for a nice, feature packed release. 
> I guess we'll know more on timing when James gets back for vacation.


did you not see his response on the list?

personally, i'd say that as we get these things updated and applied to master, 
we should also be pushing master to FGADDON...

in other words, the huge PR that was just done and advertised as available in 
the forums could already be available in FGADDON... then this next one, when it 
is completed and applied to master, update FGADDON again... in other words, 
every time we update master and call it ""ready for general use"", we should 
also update FGADDON but that's just me... later, when we branch or tag for the 
release, push once more from that branch/tag to FGADDON and we're golden...

remember, FGADDON is a moving target with developers updating it at any time 
just like using git...

2 centavos

**Comment by wlbragg** [comment]
@wkitty42 we're not in fgaddon, only fgdata.

**Comment by wkitty42** [comment]
On 07/03/2018 02:58 PM, wlbragg wrote:
> @wkitty42 we're not in fgaddon, only fgdata.


OOPS! darned fingers... been typing FGADDON so much today... but yeah... maybe i 
can get by the web interface and edit my post to correct that... if i can find 
this issue :LOL:

thanks!

**Comment by gilbertohasnofb** [comment]
> That's fine with me, we already have enough for a nice, feature packed release. I guess we'll know more on timing when James gets back for vacation.

Great, but once we have a clear deadline let's state it here and make sure everyone knows about it so that we can also make a major release for our aircraft :smile:

**Comment by wlbragg** [comment]
Per James on the dev list.

>Right, we didn't agree a date yet, and we need to. My plan is to start the logistical discussions after I (and Stuart) are back from vacation, on the 14th of July. My wild guess is we’d aim to release 2018.3.0 sometime in September - but unsure when exactly. So if you assume you have about eight or nine weeks of development time between now and the feature-freeze, I don’t think you’d be far off.

So we have some time. 

If at all possible, it would be really nice to include, tutorials, variants and previews, states and the current interior work. Of course the current interior work takes precedence over the previews. It needs to be completed before photos are taken.

**Comment by gilbertohasnofb** [comment]
> Of course the current interior work takes precedence over the previews. It needs to be completed before photos are taken.

I think we are nearly there. As soon as we finish all major work I will take care of all previews and screenshots.

**Comment by gilbertohasnofb** [comment]
@wlbragg Here is my personal list of what we could try to tackle for the next release in categories of importance:

Must solve issues
-----------------
https://github.com/c172p-team/c172p/issues/1097
https://github.com/c172p-team/c172p/issues/1035
https://github.com/c172p-team/c172p/issues/906
https://github.com/c172p-team/c172p/issues/967
https://github.com/c172p-team/c172p/issues/1137
https://github.com/c172p-team/c172p/issues/1119

Major 3D model/texture/FX issue
-------------------------------
https://github.com/c172p-team/c172p/issues/1023
https://github.com/c172p-team/c172p/issues/802
https://github.com/c172p-team/c172p/issues/1096
https://github.com/c172p-team/c172p/issues/699
https://github.com/c172p-team/c172p/issues/1124

Major non-modelling issues
--------------------------
https://github.com/c172p-team/c172p/issues/741
https://github.com/c172p-team/c172p/issues/1026

Minor 3D/texture/FX issues (wishlist)
-------------------------------------
https://github.com/c172p-team/c172p/issues/202
https://github.com/c172p-team/c172p/issues/961
https://github.com/c172p-team/c172p/issues/1098
https://github.com/c172p-team/c172p/issues/1122
https://github.com/c172p-team/c172p/issues/1143
https://github.com/c172p-team/c172p/issues/1110
https://github.com/c172p-team/c172p/issues/1009
https://github.com/c172p-team/c172p/issues/650
https://github.com/c172p-team/c172p/issues/447
https://github.com/c172p-team/c172p/issues/417
https://github.com/c172p-team/c172p/issues/464
https://github.com/c172p-team/c172p/issues/626
https://github.com/c172p-team/c172p/issues/1022
https://github.com/c172p-team/c172p/issues/1003
https://github.com/c172p-team/c172p/issues/1093
https://github.com/c172p-team/c172p/issues/1030
https://github.com/c172p-team/c172p/issues/1046
https://github.com/c172p-team/c172p/issues/881

Other minor issues to consider
------------------------------
https://github.com/c172p-team/c172p/issues/1134
https://github.com/c172p-team/c172p/issues/724
https://github.com/c172p-team/c172p/issues/306
https://github.com/c172p-team/c172p/issues/293
https://github.com/c172p-team/c172p/issues/538
https://github.com/c172p-team/c172p/issues/544
https://github.com/c172p-team/c172p/issues/1008
https://github.com/c172p-team/c172p/issues/999

This list has a bunch of really small issues that we can fix very quickly and/or be fixed in single PRs. I would say we could start with the major 3D ones, namely fixing our levers, adding the low voltage light and recreating the shadow and reflection cubes since the shape of our interior changed considerably (no headrests, higher panel, etc.).

**Comment by wlbragg** [comment]
@gilbertohasnofb wow, thanks for the detail. Looking it over it looks good and much of it should go quickly.
I plan to work on variants after you start on the previews as there may be some specific previews we want for the variants.
I also feel like the "start up state" work is vary important because it is tied so closely with the simulators GUI mechanics now. I plan to dive into that at any time (unless @onox was to miraculously beat me to it) , but it shouldn't interfere with other priorities.

**Comment by gilbertohasnofb** [comment]
Great! Do feel free to add/remove anything from the list. We can just try to tackle as much. I also added https://github.com/c172p-team/c172p/issues/789 to the list.

**Comment by gilbertohasnofb** [comment]
@wlbragg I have crossed all issues that are already fixed from [our list](https://github.com/c172p-team/c172p/issues/1108#issuecomment-402544610). We now need to update the first post with the list of fixes from that gigantic PR. Should I do it myself and you give it a look or do you want to do it yourself?

**Comment by wlbragg** [comment]
@gilbertohasnofb if you have the time you might take a pass at it and I will follow up and do the same.

**Comment by wlbragg** [comment]
FYI: a reminder of some of the new items...
I didn't list any of the bug fixes as I don't think they were really necessary to mentioned as they mostly were value added bugs we just happened top notice.

Low Voltage
New interior door handle animation
New straps for closing doors
Overhaul antenna systems
New ADF, ELT and Beacon marker antennas
Improvements to vertical stabilizer
Retopologized beacon
New animated yoke radio talk cord
Automated door closing do to wind resistance

**Comment by gilbertohasnofb** [comment]
@wlbragg Done. I think I included all the ones you pointed out, but please take a look and feel free to change whatever you want, including my wording.

**Comment by gilbertohasnofb** [comment]
@wlbragg I am crossing the issues from the list that have either been solved or postponed for another release, just to make it clear.

**Comment by wlbragg** [comment]
I saw that, good!

**Comment by onox** [comment]
@legoboyvdlp The @c172p-team/developers team would like to add you to the team :smile: Would you like to be added?

**Comment by legoboyvdlp** [comment]
Do you actually trust me given my interesting history with tabs? :joy:  

Anyway, that would very good - thank you! It will make the workflow a lot easier, without having to work with forks.

I understand the master branch is protected - i.e. the only thing that should ever be pushed there is a merge commit from another branch?

**Comment by gilbertohasnofb** [comment]
@legoboyvdlp Cool lego! This will make life easier when we want to contribute with you in a same PR, also pushing commits (just like I wanted to do with the tutorials one). And you also will get access to the developers chat :smile:

**Comment by gilbertohasnofb** [comment]
And yeah, `master` is protected and you should never ever merge your own PRs, always let someone else test and review it. You will also be able to merge PRs from now on, but do please use a lot of thought before hitting the merge button. If a PR isn't in your area of expertise or if it's a complex large one then do consider contributing with your review but not merging before more people also try it.

**Comment by legoboyvdlp** [comment]
I just added my digital clock.
